packages/@uppy/transloadit/src/index.js

```diff
@@ -212,17 +212,14 @@ export default class Transloadit extends BasePlugin {
         },
       })
 
-      const { files } = this.uppy.getState()
+      const files = this.uppy.getFiles() // []
       const updatedFiles = {}
-      fileIDs.forEach((id) => {
-        updatedFiles[id] = this.#attachAssemblyMetadata(this.uppy.getFile(id), status)
-      })
-      this.uppy.setState({
-        files: {
-          ...files,
-          ...updatedFiles,
-        },
+      files.forEach((file) => {
+        updatedFiles[file.id] = this.#attachAssemblyMetadata(file, status)
       })
+
+      this.uppy.setState({ files: updatedFiles })
```

Comment:
That looks wrong, shouldn't if stay as it was before?
```suggestion
      this.uppy.setState({
        files: {
          ...files,
          ...updatedFiles,
        },
      })
```

Reply:
`files` in this case is an array, but object is needed for `uppy.state.files`.

Reply:
But you are right. We need to save the original files as well, otherwise we’ll only have the updated ones in state, when some file might have not been selected for the assembly for some reason.

Reply:
```suggestion
      this.uppy.setState({
        files: {
          ...this.uppy.getState().files,
          ...updatedFiles,
        },
      })
```

Reply:
Hum I think both our suggestions are incorrect. It should be something like:

```js
      const files = this.uppy.getFiles().map(file => this.#attachAssemblyMetadata(file, status))
      this.uppy.setState({ files })
```

Reply:
lol, should we rename the method to `uppy.getFilesArray()` in 3.0?